src/material/core/private/ripple-loader.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {
+  ANIMATION_MODULE_TYPE,
+  ElementRef,
+  Injectable,
+  NgZone,
+  OnDestroy,
+  inject,
+} from '@angular/core';
+import {MAT_RIPPLE_GLOBAL_OPTIONS, MatRipple} from '../ripple';
+import {Platform} from '@angular/cdk/platform';
+
+/** The options for the MatRippleLoader's event listeners. */
+const eventListenerOptions = {capture: true};
+
+/** The events that should trigger the initialization of the ripple. */
+const rippleInteractionEvents = ['focus', 'click', 'mouseenter', 'touchstart'];
+
+/** The attribute attached to a component whose ripple has not yet been initialized. */
+const matRippleUninitialized = 'mat-ripple-uninitialized';
```

Comment:
should these be `mat-ripple-loader-*` to avoid possible collisions with attrs the ripple itself might set?